File: src/main/java/com/mbayou/kick4k/events/type/ChannelFollowedEvent.java
package com.mbayou.kick4k.events.type;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class ChannelFollowedEvent extends KickEvent {
    private final EventUser broadcaster;
    private final EventUser follower;

    @JsonCreator
    public ChannelFollowedEvent(@JsonProperty("broadcaster") EventUser broadcaster,
                                @JsonProperty("follower") EventUser follower) {
        this.broadcaster = broadcaster;
        this.follower = follower;
    }

    public EventUser getBroadcaster() {
        return broadcaster;
    }

    public EventUser getFollower() {
        return follower;
    }

    public static String getEventType() {
        return "channel.followed";
    }

    public static String getEventVersion() {
        return "1";
    }
}

File: src/main/kotlin/com/mbayou/kick4k/events/type/ChannelFollowedEvent.kt
package com.mbayou.kick4k.events.type

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class ChannelFollowedEvent @JsonCreator constructor(
    @JsonProperty("broadcaster") val broadcaster: EventUser,
    @JsonProperty("follower") val follower: EventUser,
) : KickEvent() {
    companion object {
        @JvmStatic
        fun getEventType(): String = "channel.followed"

        @JvmStatic
        fun getEventVersion(): String = "1"
    }
}
